api/auth/azure/azure.go

```diff
@@ -0,0 +1,240 @@
+		return nil
+	}
+}
+
+// Retrieves an access token from Managed Identities for Azure Resources
+//
+// Learn more here: https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token
+func (a *AzureAuth) getJWT() (string, error) {
+	identityEndpoint, err := url.Parse(fmt.Sprintf("%s/metadata/identity/oauth2/token", metadataEndpoint))
+	if err != nil {
+		return "", fmt.Errorf("error creating metadata URL: %w", err)
+	}
+
+	identityParameters := identityEndpoint.Query()
+	identityParameters.Add(apiVersionQueryParam, metadataAPIVersion)
+	identityParameters.Add(resourceQueryParam, a.resource)
+	identityEndpoint.RawQuery = identityParameters.Encode()
+
+	req, err := http.NewRequest("GET", identityEndpoint.String(), nil)
+	if err != nil {
+		return "", fmt.Errorf("error creating HTTP request: %w", err)
+	}
+	req.Header.Add("Metadata", "true")
+
+	client := &http.Client{
+		Timeout: clientTimeout,
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("error calling Azure token endpoint: %w", err)
+	}
+	defer resp.Body.Close()
+
+	responseBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("error reading response body from Azure token endpoint: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		var errResp errorJSON
+		_ = json.Unmarshal(responseBytes, &errResp)
+		if err != nil {
+			return "", fmt.Errorf("received error message but was unable to unmarshal its contents")
+		}
+		return "", fmt.Errorf("%s error from Azure token endpoint: %s", errResp.Error, errResp.ErrorDescription)
+	}
+
+	var r responseJSON
+	err = json.Unmarshal(responseBytes, &r)
+	if err != nil {
+		return "", fmt.Errorf("error unmarshaling response from Azure token endpoint: %w", err)
+	}
+
+	return r.AccessToken, nil
+}
+
+func getMetadata() (metadataJSON, error) {
+	metadataEndpoint, err := url.Parse(fmt.Sprintf("%s/metadata/instance", metadataEndpoint))
+	if err != nil {
+		fmt.Println("Error creating URL: ", err)
+		return metadataJSON{}, err
+	}
+
+	metadataParameters := metadataEndpoint.Query()
+	metadataParameters.Add(apiVersionQueryParam, metadataAPIVersion)
+	metadataEndpoint.RawQuery = metadataParameters.Encode()
+	req, err := http.NewRequest("GET", metadataEndpoint.String(), nil)
+	if err != nil {
+		return metadataJSON{}, fmt.Errorf("error creating HTTP Request for metadata endpoint: %w", err)
+	}
+	req.Header.Add("Metadata", "true")
+
+	client := &http.Client{
+		Timeout: clientTimeout,
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return metadataJSON{}, fmt.Errorf("error calling metadata endpoint: %w", err)
+	}
+	defer resp.Body.Close()
+
+	responseBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return metadataJSON{}, fmt.Errorf("error reading response body from metadata endpoint: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		var errResp errorJSON
+		_ = json.Unmarshal(responseBytes, &errResp)
```

Comment:
Is the error in here being ignored on purpose? If so, the err check right below would be checking for the error from another function (last assignment seems to be from `ioutil.ReadAll`)